### PR TITLE
Disable the memory64 feature in Memory64Lowering.cpp

### PR DESCRIPTION
**Pull request by tlively**
For consistency with other feature lowering passes, disable memory64 in addition
to lowering its use away. Although no other passes would introduce new uses of
memory64 at the moment, this makes the lowering pass more robust against a
future where memory64 might accidentally be reintroduced after being lowered away.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5676** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5676" target="_blank">Graphite</a> 
    * **PR #5679** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5679" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5679?utm_source=stack-comment).

File: test/lit/passes/memory64-lowering-features.wast [hunk 1]
```diff
@@ -0,0 +1,9 @@
+;; RUN: wasm-opt %s --enable-memory64 --print-features --print --memory64-lowering --print-features | filecheck %s
+
+;; Check that the --signext-lowering pass disables the signext feature.
```
Comment (kripken) [inline test/lit/passes/memory64-lowering-features.wast:3]:
```suggestion
;; Check that the --memory64-lowering pass disables the signext feature.
```

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
@tlively queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5679).

**Comment by tlively** [comment]
@tlively queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5679).

**Pull request merged**